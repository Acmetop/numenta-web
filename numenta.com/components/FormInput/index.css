/* Numenta company website source code. Copyright Â© 2016 Numenta.*/
/* Full details in LICENSE.txt, or contact us at <http://numenta.com>.*/
/* This program is free software: you can redistribute it and/or modify it under*/
/* the terms of the GNU Affero General Public License as published by the Free*/
/* Software Foundation, either version 3 of the License, or (at your option) any*/
/* later version. This program is distributed in the hope that it will be*/
/* useful, but WITHOUT ANY WARRANTY; without even the implied warranty of*/
/* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero*/
/* General Public License for more details. You should have received a copy of*/
/* the GNU Affero General Public License along with this program. If not, see*/
/* <https://www.gnu.org/licenses/agpl.html>.*/

._focus:focus {
  border: 2px solid #1aa0db;
  color: #000;
  outline: 0;
}

.large {
  /* */
}
.medium {
  max-width: 20rem;
}
.small {
  max-width: 12rem;
}

.formInput {
  composes: ba from 'tachyons';
  composes: b--black-20 from 'tachyons';
  composes: br2 from 'tachyons';
  composes: f5 from 'tachyons';
  composes: flex-auto from 'tachyons';
  composes: fw3 from 'tachyons';
  composes: input-reset from 'tachyons';
  composes: pb1 from 'tachyons';
  composes: ph2 from 'tachyons';
  composes: pt2 from 'tachyons';
  composes: silver from 'tachyons';
  composes: _focus;
}
