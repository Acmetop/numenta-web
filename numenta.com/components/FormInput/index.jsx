// Numenta company website source code. Copyright Â© 2016 Numenta.
// Full details in LICENSE.txt, or contact us at <http://numenta.com>.
// This program is free software: you can redistribute it and/or modify it under
// the terms of the GNU Affero General Public License as published by the Free
// Software Foundation, either version 3 of the License, or (at your option) any
// later version. This program is distributed in the hope that it will be
// useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
// General Public License for more details. You should have received a copy of
// the GNU Affero General Public License along with this program. If not, see
// <https://www.gnu.org/licenses/agpl.html>.

import classNames from 'classnames'
import React from 'react'

import styles from './index.css'


const FormInput = ({name, onChange, placeholder, stretch, type, value}) => {
  const classes = [styles.formInput]

  if (stretch) classes.push(styles[stretch])

  return (
    <input
      className={classNames(...classes)}
      id={name}
      name={name}
      onChange={onChange}
      placeholder={placeholder}
      type={type}
      value={value}
    />
  )
}

FormInput.propTypes = {
  name: React.PropTypes.string.isRequired,
  onChange: React.PropTypes.func,
  placeholder: React.PropTypes.string,
  stretch: React.PropTypes.oneOf(['small', 'medium', 'large']),
  type: React.PropTypes.string.isRequired,
  value: React.PropTypes.string,
}

FormInput.defaultProps = {
  stretch: 'large',
}

export default FormInput
