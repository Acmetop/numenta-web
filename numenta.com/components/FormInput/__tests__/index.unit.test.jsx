import React from 'react'
import renderer from 'react-test-renderer'

import FormInput from '../../FormInput'


describe('FormInput React component', () => {

  it('Renders correctly', () => {
    const component = renderer.create(
      <FormInput name="field" type="text" value="contents" />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
