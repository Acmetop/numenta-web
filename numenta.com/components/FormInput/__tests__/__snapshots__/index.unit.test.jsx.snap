exports[`FormInput React component Renders correctly 1`] = `
<input
  className="formInput large"
  id="field"
  name="field"
  onChange={undefined}
  placeholder={undefined}
  type="text"
  value="contents" />
`;
